src/compile/kinds/compiler.ts

```diff
@@ -1,14 +1,155 @@
+import axios from "axios";
+import { spawn } from "child_process";
+import fse from "fs-extra";
+import path from "path";
+import { satisfies } from "semver";
+import * as stream from "stream";
+import { promisify } from "util";
+import { CompilerKind, CompilerVersions } from "..";
 import { assert } from "../../misc";
 import { SolcInput } from "../input";
 import { isExact } from "../version";
+import {
+    BINARIES_URL,
+    CACHE_DIR,
+    getCompilerMDForPlatform,
+    getCompilerPrefixForOs,
+    isSubDir
+} from "./md";
+
+const solc0426 = require("solc-0.4.26");
+const solc0811 = require("solc-0.8.11");
 
 export abstract class Compiler {
-    constructor(public readonly version: string) {
-        assert(
-            isExact(version),
-            "Version string must contain exact SemVer-formatted version without any operators"
-        );
-    }
+    constructor(public readonly version: string, public readonly path: string) {}
 
     abstract compile(inputJson: SolcInput): Promise<any>;
 }
+
+class NativeCompiler extends Compiler {
+    async compile(input: SolcInput): Promise<any> {
+        const child = spawn(this.path, ["--standard-json"], {});
+
+        return new Promise((resolve, reject) => {
+            child.stdin.write(JSON.stringify(input), "utf-8");
+            child.stdin.end();
+
+            let stdout = "";
+            let stderr = "";
+
+            child.stdout.on("data", (data) => {
+                stdout += data;
+            });
+
+            child.stderr.on("data", (data) => {
+                stderr += data;
+            });
+
+            child.on("close", (code) => {
+                if (code !== 0) {
+                    reject(`Compiler exited with code ${code}, stderr: ${stderr}`);
+                    return;
+                }
+
+                if (stderr !== "") {
+                    reject(`Compiler exited with non-empty stderr: ${stderr}`);
+                    return;
+                }
+
+                let outJson: any;
+
+                try {
+                    outJson = JSON.parse(stdout);
+                } catch (e) {
+                    reject(e);
+                    return;
+                }
+
+                resolve(outJson);
+            });
+        });
+    }
+}
+
+export class WasmCompiler extends Compiler {
+    async compile(input: SolcInput): Promise<any> {
+        const module = require(this.path);
+
+        if (satisfies(this.version, "^0.4.13")) {
+            const wrappedModule = solc0426.setupMethods(module);
```

Comment:
It sucks that we have to keep a dependency on 0.4.13 just for the setupMethods call, since that will still keep those vulnerable dependencies that dependbot complains about. I wonder if there is some other workaround?

Reply:
As suggested [above](https://github.com/ConsenSys/solc-typed-ast/pull/92#issuecomment-1034976261) we can consider to mitigate via opening an issue in SolcJS repo. I will try to make minimal reproducible example  to see if this is really an issue.

Reply:
This is actuall a bug. I didn't updated input composition to use logic for Solc 0.5.* and above.

Reply:
Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/92/commits/243a46a7b1732f537208d3f22a4c0c00c5729d0e